Mathlib/Algebra/Exact.lean

```diff
@@ -35,30 +36,106 @@ variable (f : M → N) (g : N → P)
   `g y = 0` iff `y` belongs to the image of `f` -/
 def Exact [Zero P] : Prop := ∀ y, g y = 0 ↔ y ∈ Set.range f
 
+variable {f g}
+
+lemma Exact.apply_apply_eq_zero [Zero P] (h : Exact f g) (x : M) :
+    g (f x) = 0 := (h _).mpr <| Set.mem_range_self _
+
 lemma Exact.comp_eq_zero [Zero P] (h : Exact f g) : g.comp f = 0 :=
-  funext fun _ => (h _).mpr <| Set.mem_range_self _
+  funext h.apply_apply_eq_zero
+
+lemma Exact.of_comp_eq_zero_of_eq_zero_imp_mem_range [Zero P]
```

Comment:
This is a bit mouthful. Maybe `Exact.of_comp_eq_zero_of_mem_range` or even `Exact.of_comp_of_mem_range`.